src/unix/groups.rs

```diff
@@ -0,0 +1,59 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::common::Gid;
+
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+use crate::Group;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub(crate) struct GroupInner {
+    pub(crate) gid: Gid,
+    pub(crate) name: String,
+}
+
+impl GroupInner {
+    pub(crate) fn new(gid: Gid, name: String) -> Self {
+        Self { gid, name }
+    }
+
+    pub(crate) fn id(&self) -> &Gid {
+        &self.gid
+    }
+
+    pub(crate) fn name(&self) -> &str {
+        &self.name
+    }
+}
+
+// Not used by mac.
+#[cfg(not(any(target_os = "macos", target_os = "ios")))]
+pub(crate) fn get_groups(groups: &mut Vec<Group>) {
+    use std::fs::File;
+    use std::io::Read;
+
+    #[inline]
+    fn parse_id(id: &str) -> Option<u32> {
+        id.parse::<u32>().ok()
+    }
+
+    groups.clear();
+
+    let mut s = String::new();
+
+    let _ = File::open("/etc/group").and_then(|mut f| f.read_to_string(&mut s));
```

Comment:
Why not using `getgrent` for freebsd and linux? The function exists in both cases and that would allow to have one common code for all unix platforms.

Reply:
Because the `users` module reads `/etc/passwd` on FreeBSD and Linux, so I copied the pattern. Fixed in last commit.